resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -105,6 +131,44 @@ private[spark] class KubernetesClusterSchedulerBackend(
 
   private val initialExecutors = getInitialTargetExecutorNumber(1)
 
+  private val podAllocationInterval =
+    if (conf.get(KUBERNETES_ALLOCATION_BATCH_DELAY) > 0) {
```

Comment:
There's probably no need to make this an option - just assign `podAllocationInterval` directly, and then check the variable directly + throw the SparkException immediately afterwards.